numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -649,8 +649,7 @@ fromstring_null_term_c_api(PyObject *dummy, PyObject *byte_obj)
     if (string == NULL) {
         return NULL;
     }
-    descr = PyArray_DescrNewFromType(NPY_FLOAT64);
-    return PyArray_FromString(string, -1, descr, -1, " ");
+    return PyArray_FromString(string, -1, NULL, -1, " ");
```

Comment:
Allow PyArray_FromString to allocate the descriptor: https://github.com/numpy/numpy/blob/c65bc212ec1987caefba0ea7efe6a55803318de9/numpy/core/src/multiarray/ctors.c#L3743-L3746